### PR TITLE
Tests for anons on default wiki get stuck on GDPR modal.

### PR DESCRIPTION
**Pull request by sczerwinski-wikia**

### CODE REVIEW

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [PageObjectLogging.java#L172](https://github.com/Wikia/selenium-tests/blob/aebdb0f25e84521dc591b6c6ee2b756cd74128b8/src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java#L172): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MINOR][MINOR] [PageObjectLogging.java#L288](https://github.com/Wikia/selenium-tests/blob/aebdb0f25e84521dc591b6c6ee2b756cd74128b8/src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java#L288): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS134)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [PageObjectLogging.java#L172](https://github.com/Wikia/selenium-tests/blob/191ffaefe3a15183e800b12eabfcf9b11dd6e31a/src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java#L172): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MINOR][MINOR] [PageObjectLogging.java#L288](https://github.com/Wikia/selenium-tests/blob/191ffaefe3a15183e800b12eabfcf9b11dd6e31a/src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java#L288): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS134)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsMedrecReloading.java [hunk 1]
```diff
@@ -26,6 +26,11 @@ public void regularFloatingMedrecIsReloadingWithRecirculationModule() {
     AdsBaseObject ads = new AdsBaseObject(driver, page.getUrl());
 
     ads.scrollToPosition(RECIRCULATION_SELECTOR);
+    try {
+      Thread.sleep(10000);
+    } catch (InterruptedException e) {
+      e.printStackTrace();
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsMedrecReloading.java]: CRITICAL Use a logger to log this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1148)

File: src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java [hunk 2]
```diff
@@ -403,6 +390,41 @@ public void afterNavigateTo(String url, WebDriver driver) {
...
+        driver.manage().addCookie(
+            new Cookie("tracking-opt-in-status", "accepted", ".wikia.com", "/",
+                       new Date(new DateTime().plusYears(10).getMillis())));
+        driver.manage().addCookie(new Cookie("tracking-opt-in-status", "accepted"));
+      } else if (userOptedOut) {
+        driver.manage().addCookie(
+            new Cookie("tracking-opt-in-status", "rejectedgit", ".wikia.com", "/",
+                       new Date(new DateTime().plusYears(10).getMillis())));
+      }
+//      driver.navigate().refresh();
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java]: MAJOR This block of commented-out lines of code should be removed. [rule](http://sonar-s4/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 7 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 5 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

#### 5 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [PageObjectLogging.java#L172](https://github.com/Wikia/selenium-tests/blob/ed056fcab70947383635d5ccae1042e3d1d26741/src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java#L172): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [TestAdsMedrecReloading.java#L77](https://github.com/Wikia/selenium-tests/blob/ed056fcab70947383635d5ccae1042e3d1d26741/src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsMedrecReloading.java#L77): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [TestAdsMedrecReloading.java#L77](https://github.com/Wikia/selenium-tests/blob/ed056fcab70947383635d5ccae1042e3d1d26741/src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsMedrecReloading.java#L77): Either remove or fill this block of code. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS00108)
1. ![MAJOR][MAJOR] [TestAdsMedrecReloading.java#L97](https://github.com/Wikia/selenium-tests/blob/ed056fcab70947383635d5ccae1042e3d1d26741/src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsMedrecReloading.java#L97): Either remove or fill this block of code. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS00108)
1. ![MINOR][MINOR] [PageObjectLogging.java#L288](https://github.com/Wikia/selenium-tests/blob/ed056fcab70947383635d5ccae1042e3d1d26741/src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java#L288): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS134)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsMedrecReloading.java [hunk 3]
```diff
@@ -25,6 +25,11 @@ public void regularFloatingMedrecIsReloadingWithRecirculationModule() {
     Page page = new Page("project43", "SyntheticTests/LongPage");
     AdsBaseObject ads = new AdsBaseObject(driver, page.getUrl());
 
+    try {
+      Thread.sleep(10000);
+    } catch (InterruptedException e) {
+      e.printStackTrace();
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsMedrecReloading.java]: CRITICAL Use a logger to log this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1148)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 7 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 5 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

#### 5 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [PageObjectLogging.java#L172](https://github.com/Wikia/selenium-tests/blob/21ca4b389d73f55e233d75eee3122fbaa868bcc3/src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java#L172): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [TestAdsMedrecReloading.java#L77](https://github.com/Wikia/selenium-tests/blob/21ca4b389d73f55e233d75eee3122fbaa868bcc3/src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsMedrecReloading.java#L77): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [TestAdsMedrecReloading.java#L77](https://github.com/Wikia/selenium-tests/blob/21ca4b389d73f55e233d75eee3122fbaa868bcc3/src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsMedrecReloading.java#L77): Either remove or fill this block of code. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS00108)
1. ![MAJOR][MAJOR] [TestAdsMedrecReloading.java#L97](https://github.com/Wikia/selenium-tests/blob/21ca4b389d73f55e233d75eee3122fbaa868bcc3/src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsMedrecReloading.java#L97): Either remove or fill this block of code. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS00108)
1. ![MINOR][MINOR] [PageObjectLogging.java#L288](https://github.com/Wikia/selenium-tests/blob/21ca4b389d73f55e233d75eee3122fbaa868bcc3/src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java#L288): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS134)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsMedrecReloading.java [hunk 4]
```diff
@@ -25,6 +25,9 @@ public void regularFloatingMedrecIsReloadingWithRecirculationModule() {
     Page page = new Page("project43", "SyntheticTests/LongPage");
     AdsBaseObject ads = new AdsBaseObject(driver, page.getUrl());
 
+    driver.manage().getCookies().stream().forEach(c -> System.out.println(c.getName() + " = " + c.getValue() +
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsMedrecReloading.java]: MAJOR Replace this usage of System.out or System.err by a logger. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS106)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 7 issues
* ![MAJOR][MAJOR] 6 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

#### 5 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [PageObjectLogging.java#L172](https://github.com/Wikia/selenium-tests/blob/2239a548586cd67f4bee8d940ea4d44e93565277/src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java#L172): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [TestAdsMedrecReloading.java#L75](https://github.com/Wikia/selenium-tests/blob/2239a548586cd67f4bee8d940ea4d44e93565277/src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsMedrecReloading.java#L75): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [TestAdsMedrecReloading.java#L75](https://github.com/Wikia/selenium-tests/blob/2239a548586cd67f4bee8d940ea4d44e93565277/src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsMedrecReloading.java#L75): Either remove or fill this block of code. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS00108)
1. ![MAJOR][MAJOR] [TestAdsMedrecReloading.java#L95](https://github.com/Wikia/selenium-tests/blob/2239a548586cd67f4bee8d940ea4d44e93565277/src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsMedrecReloading.java#L95): Either remove or fill this block of code. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS00108)
1. ![MINOR][MINOR] [PageObjectLogging.java#L288](https://github.com/Wikia/selenium-tests/blob/2239a548586cd67f4bee8d940ea4d44e93565277/src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java#L288): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS134)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java [hunk 5]
```diff
@@ -337,34 +350,8 @@ public void afterNavigateTo(String url, WebDriver driver) {
...
-          driver.manage().addCookie(
-              new Cookie("tracking-opt-in-status", "accepted", ".wikia.com", "/",
-                         new Date(new DateTime().plusYears(10).getMillis())));
-        } else if (userOptedOut) {
-          driver.manage().addCookie(
-              new Cookie("tracking-opt-in-status", "rejectedgit", ".wikia.com", "/",
-                         new Date(new DateTime().plusYears(10).getMillis())));
-        }
-      }
+//      handleGdprModalWithCookie();
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java]: MAJOR This block of commented-out lines of code should be removed. [rule](http://sonar-s4/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

File: src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java [hunk 6]
```diff
@@ -403,6 +390,41 @@ public void afterNavigateTo(String url, WebDriver driver) {
     logJSError();
   }
 
+  /**
+   * All of tests should be executed as an user who opted in (agreed) on using ads tracking,
+   * unless you change it in @Execute annotation.
+   * Manually user would need to click 'agree' in the tracking opt in modal.
+   */
+  private void handleGdprModalWithCookie() {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java]: MAJOR Remove this unused private "handleGdprModalWithCookie" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 9 issues
* ![MAJOR][MAJOR] 8 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

#### 5 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [PageObjectLogging.java#L172](https://github.com/Wikia/selenium-tests/blob/6c68f83e89dad59249d6a698a26ee6fc4f77a80a/src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java#L172): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [TestAdsMedrecReloading.java#L75](https://github.com/Wikia/selenium-tests/blob/6c68f83e89dad59249d6a698a26ee6fc4f77a80a/src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsMedrecReloading.java#L75): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [TestAdsMedrecReloading.java#L75](https://github.com/Wikia/selenium-tests/blob/6c68f83e89dad59249d6a698a26ee6fc4f77a80a/src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsMedrecReloading.java#L75): Either remove or fill this block of code. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS00108)
1. ![MAJOR][MAJOR] [TestAdsMedrecReloading.java#L95](https://github.com/Wikia/selenium-tests/blob/6c68f83e89dad59249d6a698a26ee6fc4f77a80a/src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsMedrecReloading.java#L95): Either remove or fill this block of code. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS00108)
1. ![MINOR][MINOR] [PageObjectLogging.java#L288](https://github.com/Wikia/selenium-tests/blob/6c68f83e89dad59249d6a698a26ee6fc4f77a80a/src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java#L288): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS134)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by ludwikkazmierczak** [comment]
It's little old ;)

**Pull request closed without merging**